functions.php

```diff
@@ -20,4 +17,32 @@ function connect_db($collectionName)
     JOIN `TVshows` ON `cartoons`.`TVshow_id` = `TVshows`.`id`;');
     $query->execute();
     return $query->fetchAll();
+}
+
+/*
+ * This takes the given array $cartoons, ensures it is a string, and concatenates each variable, connecting
+ * each bit of data from the database, to give the $character an image, name, TV show name and IQ
+ *
+ * @param $cartoons the array, from the database, which is being passed in this function
+ *
+ * @return this returns the variable $characterHtmlElementsString which is a concatenated string with each separate stat
+ * about each character
+ *
+ */
+
+function eachCharacter(array $cartoons): string
+{
+    $characterHtmlElementsString = '';
+    foreach ($cartoons as $character) {
+        $characterImageElement = '<img src="' . $character['image'] . '" alt="Picture of ' . $character['character_name'] . '">';
+        $characterNameElement = $character['character_name'];
+        $characterTvShowElement = $character['name'];
+        $characterIqElement = $character['IQ'];
+        $characterHtmlElement = '<div class="container">' . $characterImageElement . "<h2>Name: " . $characterNameElement . '</h2>' . "<h2>TV Show: " . $characterTvShowElement . '</h2>' . "<h2>IQ: " . $characterIqElement . '</h2>' . '</div>';
+        $characterHtmlElementsString .= $characterHtmlElement;
+        if (!isset($character['character_name']) || !isset($character['IQ']) || !isset($character['name']) || !isset($character['image'])) {
+            return false;
```

Comment:
Because you're specified that the return type must be a string this will give you issues with your unit testing, you can correct this by returning an empty string ''.